packages/server/postgres/migrations/2025-04-17T13:33:36.200Z_RBAC.ts

```diff
@@ -0,0 +1,261 @@
+FOR EACH ROW EXECUTE FUNCTION "updateUserPageAccess"();
+
+
+-- UPDATE CACHE FROM PageTeamAccess
+CREATE OR REPLACE FUNCTION "updateTeamPageAccess"() RETURNS TRIGGER AS $$
+DECLARE
+  "_userId" VARCHAR;
+BEGIN
+  FOR "_userId" IN SELECT "userId" FROM "TeamMember" WHERE "teamId" = NEW."teamId" LOOP
+  PERFORM "updatePageAccess"("_userId", NEW."pageId");
+  END LOOP;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE TRIGGER trg_team_page_access
+AFTER INSERT OR UPDATE OR DELETE ON "PageTeamAccess"
+FOR EACH ROW EXECUTE FUNCTION "updateTeamPageAccess"();
+
+
+-- UPDATE CACHE FROM PageOrganizationAccess
+CREATE OR REPLACE FUNCTION "updateOrganizationPageAccess"() RETURNS TRIGGER AS $$
+DECLARE
+  "_userId" VARCHAR;
+BEGIN
+  FOR "_userId" IN SELECT "userId" FROM "OrganizationUser" WHERE "orgId" = NEW."orgId" LOOP
+  PERFORM "updatePageAccess"("_userId", NEW."pageId");
+  END LOOP;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE TRIGGER trg_org_page_access
+AFTER INSERT OR UPDATE OR DELETE ON "PageOrganizationAccess"
+FOR EACH ROW EXECUTE FUNCTION "updateOrganizationPageAccess"();
+
+
+-- HANDLE JOIN/LEAVE TEAM
+CREATE OR REPLACE FUNCTION "updateTeamPageAccessByTeamMember"() RETURNS TRIGGER AS $$
+DECLARE
+  "_pageId" INT;
+BEGIN
+  IF (TG_OP = 'INSERT') OR (TG_OP = 'UPDATE' AND NEW."isNotRemoved" != OLD."isNotRemoved") THEN
+    FOR "_pageId" IN SELECT "pageId" FROM "PageTeamAccess" WHERE "teamId" = NEW."teamId" LOOP
+      PERFORM "updatePageAccess"(NEW."userId", "_pageId");
+    END LOOP;
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE TRIGGER trg_team_member_update_team_page_access
+AFTER INSERT OR UPDATE ON "TeamMember"
+FOR EACH ROW EXECUTE FUNCTION "updateTeamPageAccessByTeamMember"();
+
+
+-- HANDLE ARCHIVE TEAM
+CREATE OR REPLACE FUNCTION "removePageAccessOnTeamArchive"() RETURNS TRIGGER AS $$
+BEGIN
+  IF NEW."isArchived" = TRUE AND (OLD."isArchived" != TRUE) THEN
+    DELETE FROM "PageTeamAccess"
+    WHERE "teamId" = NEW."id";
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE TRIGGER trg_team_archived_remove_page_access
+AFTER UPDATE ON "Team"
+FOR EACH ROW
+EXECUTE FUNCTION "removePageAccessOnTeamArchive"();
+
+
+-- HANDLE JOIN/LEAVE ORG
+CREATE OR REPLACE FUNCTION "updateOrgPageAccessByOrgUser"() RETURNS TRIGGER AS $$
+DECLARE
+  "_pageId" INT;
+BEGIN
+  IF (TG_OP = 'INSERT') OR (TG_OP = 'UPDATE' AND NEW."removedAt" IS DISTINCT FROM OLD."removedAt") THEN
+    FOR "_pageId" IN SELECT "pageId" FROM "PageOrganizationAccess" WHERE "orgId" = NEW."orgId" LOOP
+      PERFORM "updatePageAccess"(NEW."userId", "_pageId");
+    END LOOP;
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE TRIGGER trg_org_user_update_org_page_access
+AFTER INSERT OR UPDATE ON "OrganizationUser"
+FOR EACH ROW EXECUTE FUNCTION "updateOrgPageAccessByOrgUser"();
+`.execute(db)
+}
+
+export async function down(db: Kysely<any>): Promise<void> {
+  await Promise.all([
```

Comment:
-1 we need to drop all those triggers and functions as well, don't we?

Reply:
yep!